iceoryx_utils/source/posix_wrapper/timer.cpp

```diff
@@ -69,32 +69,46 @@ void Timer::OsTimer::callbackHelper(sigval data)
 
     auto& callbackHandle = OsTimer::s_callbackHandlePool[index];
 
-    // small optimazition to not lock the mutex if the callback handle is not valid anymore
+    // small optimization to not lock the mutex if the callback handle is not valid anymore
     if (descriptor != callbackHandle.m_descriptor.load(std::memory_order::memory_order_relaxed))
     {
         return;
     }
 
-    std::lock_guard<std::mutex> lock(callbackHandle.m_accessMutex);
-    if (!callbackHandle.m_inUse.load(std::memory_order::memory_order_relaxed))
-    {
-        return;
-    }
+    uint64_t currentCycle = (callbackHandle.m_timerType == TimerType::SOFT_TIMER)
```

Comment:
there is still a race and under some conditions a callback execution might be skipped
- timer triggers and executes the callback
    - the callbacks takes always almost as much time as the interval time
- the callback is executed, but the mutex is still locked
- the timer triggers again and tries to get the lock
    - since it cannot get the lock, the callback is not executed
- the previous timer invokation continues and unlocks the mutex
- a callback execution is skipped

-> there needs to be a loop around the mutex until currentCycle equals callbackExecutionCycle. This loop must only be executed if tryLock doesn't fail